rse-py-scripting.Rmd

```diff
@@ -103,42 +133,461 @@ optional arguments:
   -h, --help  show this help message and exit
 ```
 
-and issues errors when users give the program invalid arguments:
+As the `--help` hint shows,
+`argparse` can handle [long options][long-option-shell] as well as traditional single-letter options;
+the online help has examples of this and much more.
+
+## How can I configure common options for a program? {#rse-py-scripting-configure}
+
+A program that does exactly the same thing every time we run it
+isn't as useful as one that can work on different files
+or analyze data with different thresholds.
+Software of all kinds needs to be controlled;
+some things change more often than others,
+so we need a simple, uniform way to specify some options and leave others alone.
+
```

Comment:
The paragraph above reads like the sections preceding it never happened. It might be better to start this section by saying that command line options (which were introduced in the previous section) are only one level of / option for configuration. There's also job-specific, user-specific and system-wide configuration files...

Reply:
Good point - paragraph removed and intro amended.